aiida/orm/nodes/node.py

```diff
@@ -674,10 +674,28 @@ def put_object_from_file(self, path, key, mode='w', encoding='utf8', force=False
         :param path: absolute path of file whose contents to copy to the repository
         :param key: fully qualified identifier for the object within the repository
         :param mode: the file mode with which the object will be written
+            Deprecated: will be removed in `v1.1.0`
```

Comment:
hehe, how was this thing with increasing the **major** version number for api changes? ;-)

Reply:
Honestly, it wasn't me :) I put `v2.0.0` but @giovannipizzi strongly suggested to put `v1.1.0`. He will be my witness

Reply:
So @giovannipizzi , what is the reasoning here?
We say 1.1 means "only minor little API changes"?
What is minor and what is major?

Reply:
The reasoning is that if, in 1.0.0, something is already deprecated, people shouldn't use it (and if they have been using it in a beta, well, they know it's a beta and things can still change - we still guarantee that when we release 1.0 things work, and they have time to adapt before 1.1).

If we deprecated something in 1.0.1 or later instead, I think it has to be changed in 2.0.

Reply:
I remember that we said the API would be frozen from v1.0.0b1 onwards unless we discover real problems.
This was communicated to plugin developers as a reson why it makes sense to invest the time for the migration now, and is mentioned in the roadmap https://github.com/aiidateam/aiida-core/wiki/AiiDA-release-roadmap

I don't think this change here qualifies as a big problem.
What is the issue with leaving the parameters (deprecated) until 2.0.0?

I know this is a rather small issue here, but I think this is a matter of principle.
In my view, if a plugin breaks in aiida 1.1 that worked with aiida 1.0.0b1 it is our fault, not the fault of the developer.

Reply:
I don't agree, but since it's not critical it's ok to put 2.0 here